docs/reference/modules/node.asciidoc

```diff
@@ -360,10 +360,14 @@ To create a dedicated {ml} node, set:
 node.roles: [ ml, remote_cluster_client] <1>
 xpack.ml.enabled: true <2>
 ----
-<1> The `remote_cluster_client` role is optional but strongly recommended.
+<1> The `remote_cluster_client` role is optional but strongly recommended on all {ml} nodes.
 Otherwise, {ccs} fails when used in {ml} jobs or {dfeeds}. See <<remote-node>>.
 <2> The `xpack.ml.enabled` setting is enabled by default.
 
+WARNING: The `remote_cluster_client` role is required on all master-eligible nodes in case
+you are using {ccs}.
+Otherwise, {ccs} fails when used in {ml} jobs or {dfeeds} against remote indices.
```

Comment:
```suggestion
Otherwise, the job or {dfeed} fails when it tries to access remote indices.
```

Reply:
The ML job will not start at all. The master doesn't access the remote indices, but it needs to validate license & permissions.

Reply:
More like

> If you use CCS in anomaly detection jobs, the `remote_cluster_client` role is required on all master-eligible nodes or the datafeed will not start. It is also required on all data nodes, or CCS jobs will fail to query the required data.

In short, it should just be enabled on all data and master nodes :)

Reply:
WOOPS! Forget that I ever mentioned data nodes. I meant `ML` nodes.

Reply:
Thanks, that makes more sense. I've edited my suggestion.